starter-code/pizza.js

```diff
@@ -17,69 +17,205 @@ var state = {
     if (state.pepperonni) {
-      $pep.style.visibility = "visible";
+      pep.style.visibility = "visible";
+    } else {
+      pep.style.visibility = "hidden";
     }
-    else {
-      $pep.style.visibility = "hidden";
-    }
-  })
+  });
 }
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll(".mushroom").forEach(function(mush) {
+    if (state.mushrooms) {
+      mush.style.visibility = "visible";
+    } else {
+      mush.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGreenPeppers() {
-  // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll(".green-pepper").forEach(function(gPep) {
+    if (state.greenPeppers) {
+      gPep.style.visibility = "visible";
+    } else {
+      gPep.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderWhiteSauce() {
-  // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">
+  if (state.whiteSauce) {
+    document.querySelector(".sauce").classList.add("sauce-white");
+  } else {
+    document.querySelector(".sauce").classList.remove("sauce-white");
+  }
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  if (state.glutenFreeCrust) {
+    document.querySelector(".crust").classList.add("crust-gluten-free");
+  } else {
+    document.querySelector(".crust").classList.remove("crust-gluten-free");
+  }
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  // document.querySelectorAll(".btn").forEach(function(btn) {
+  //   btn.classList.toggle("active");
+  // });
+
+  if (state.pepperonni) {
+    document.querySelector(".btn-pepperonni").classList.add("active");
+  } else {
+    document.querySelector(".btn-pepperonni").classList.remove("active");
+  }
+
+  if (state.mushrooms) {
+    document.querySelector(".btn-mushrooms").classList.add("active");
+  } else {
+    document.querySelector(".btn-mushrooms").classList.remove("active");
+  }
+
+  if (state.greenPeppers) {
+    document.querySelector(".btn-green-peppers").classList.add("active");
+  } else {
+    document.querySelector(".btn-green-peppers").classList.remove("active");
+  }
+
+  if (state.whiteSauce) {
+    document.querySelector(".btn-sauce").classList.add("active");
+  } else {
+    document.querySelector(".btn-sauce").classList.remove("active");
+  }
+
+  if (state.glutenFreeCrust) {
+    document.querySelector(".btn-crust").classList.add("active");
+  } else {
+    document.querySelector(".btn-crust").classList.remove("active");
+  }
 }
 
 function renderPrice() {
   // Iteration 4: change the HTML of `<aside class="panel price">`
+
+  let priceList = document.querySelector(".panel.price>ul");
+  priceList.innerHTML = "";
+
```

Comment:
same comment as above. In order to reduce your code, try to make a loop as you need to do the same for each ingredient.